sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CanonicalizeSuite.scala

```diff
@@ -95,4 +96,19 @@ class CanonicalizeSuite extends SparkFunSuite {
     val castWithTimeZoneId = Cast(literal, LongType, Some(TimeZone.getDefault.getID))
     assert(castWithTimeZoneId.semanticEquals(cast))
   }
+
+  test("SPARK-32927: Bitwise operations are commutative") {
```

Comment:
Could you add more test cases, e.g., non-deterministic exprs, literals, and mixed cases `BitwiseOr`/`BitwiseAnd`/`BitwiseXor`?